spec/scenarios/charge_models/prorated_graduated_spec.rb

```diff
@@ -0,0 +1,399 @@
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '2' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 16)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '5' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 20)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '-6' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 25)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '10' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 26)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '4' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 30)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '60' },
+              },
+            )
+
+            fetch_current_usage(customer:)
+            expect(json[:customer_usage][:amount_cents].round(2)).to eq(19033)
+            expect(json[:customer_usage][:total_amount_cents].round(2)).to eq(19033)
+            expect(json[:customer_usage][:charges_usage][0][:units]).to eq('75.0')
+          end
+
+          travel_to(DateTime.new(2023, 10, 1)) do
+            Subscriptions::BillingService.new.call
+
+            perform_all_enqueued_jobs
+
+            subscription = customer.subscriptions.first
+            invoice = subscription.invoices.first
+
+            aggregate_failures do
+              expect(invoice.total_amount_cents).to eq(19033)
+              expect(subscription.reload.invoices.count).to eq(1)
+            end
+          end
+
+          travel_to(DateTime.new(2023, 10, 5)) do
+            fetch_current_usage(customer:)
+            expect(json[:customer_usage][:amount_cents].round(2)).to eq(37000)
+            expect(json[:customer_usage][:total_amount_cents].round(2)).to eq(37000)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)